site-scrapers/Harrington/index.js

```diff
@@ -1,100 +1,255 @@
+/* eslint-disable no-unused-vars */
+/* eslint-enable no-unused-vars */
+
+/**
+ * The notificationService provides functionality to
+ * notify s3 and Slak of changes to availability.
+ *
+ * Replace with mocks in unit testing.
+ *
+ * Not used at the moment, but retained in case there's a future
+ * need to capture HTML due to changes in the website.
+ */
+let notificationService = function defaultNotificationService() {
+    return {
+        async handleAvailabilityContentUpdate(page, site) {
+            return notifyAvailabilityContentUpdate(page, site);
+        },
+    };
+};
+
+async function ScrapeWebsiteData(browser, pageService, site) {
+    console.log(`${site.name} starting to scrape`);
+    const page = await pageService.getHomePage(browser, site);
+
+    // Initialize results to no availability
+    const results = {
+        availability: {},
+        hasAvailability: false,
+    };
+
+    /*
+     monthCount is set in config.js; limit it to only a few months because this style
+     of Acuity Scheduling calendar is endless going forward in time. There is
+     no month chooser drop-down to grab a list of months.
+     */
+    for (const key of new Array(monthCount).keys()) {
+        // Don't advance the calendar if it's the first month
+        if (key > 0) {
+            await pageService.getNextMonthCalendar(page, site);
+        }
+        const dailySlotsForMonth = await getDailyAvailabilityCountsForMonth(
+            page
+        );
+        // Add all day objects to results.availability
+        dailySlotsForMonth.forEach((value, key) => {
+            results.availability[key] = {
+                numberAvailableAppointments: value,
+                hasAvailability: !!value,
+            };
+        });
+    }
+
+    results.hasAvailability = !!Object.keys(results.availability).length;
+
+    console.log(`${site.name} scrape ending`);
+
     return results;
 }
-*/
+
+/**
+ * Each slot is respresented by the following HTML:
+ *
+ *    &lt;input type="radio"
+ *        class="time-selection"
+ *        name="time[]"
+ *        data-qa="select-day-29-slot-0"
+ *        data-readable-date="2021-03-29"
+ *        value="2021-03-29 11:45"
+ *        id="appt1617032700"
+ *        aria-labelledby="lbl_appt1617032700"&gt;
+ *
+ * Select for ".time-selection" and evaluate the "data-readable-date" attribute
+ * to get the date. Appointment time (in "value" attribute) isn't needed.
+ *
+ * Map date from 'yyyy-mm-dd' to 'm/d/yyyy'.
+ *
+ * Group by date, accumulating date count, into a Map: key = date, value = slot count for date.
+ *
+ * @param {*} page
+ * @returns Map of availability keyed by date. Guaranteed not null, returning at least an empty map.
+ */
+async function getDailyAvailabilityCountsForMonth(page) {
+    function reformatDate(date) {
+        const dateObj = new Date(date + "T00:00:00");
+        return new Intl.DateTimeFormat("en-US").format(dateObj);
```

Comment:
Just an FYI, we have moment included in the repo. It might provide slightly kinder syntax to formatting the date time.